### PR TITLE
Some lewd fixes and feature

### PR DESCRIPTION
**Pull request by AnywayFarus**
## About The Pull Request

that PR should fix the possibility of climax, because earlier, by mistake of the developer, cum shoots on you, and the sheath was taken in any scenario due to an incorrect check of the variable.
Also, cum on facecomponents are not just unavailable feature, and can be used in the game via an Climax verb if your partner has prefs on for it. The indelibility of facial has also been fixed.

## How This Contributes To The Skyrat Roleplay Experience

More working mechanical interaction in the 

## Changelog

:cl: Vishenka0704
add: Cum on face action in Climax verb
fix: now you cum on chosen target, not on you.
fix: now the choice of sheath in the menu is only available with those who have sheath
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 1]
```diff
@@ -455,22 +455,24 @@
...
-						var/obj/item/organ/external/genital/penis/target_penis = getorganslot(ORGAN_SLOT_PENIS)
+						var/obj/item/organ/external/genital/vagina/target_vagina = target_human.getorganslot(ORGAN_SLOT_VAGINA)
+						var/obj/item/organ/external/genital/anus/target_anus = target_human.getorganslot(ORGAN_SLOT_ANUS)
+						var/obj/item/organ/external/genital/penis/target_penis = target_human.getorganslot(ORGAN_SLOT_PENIS)
 
 						var/list/target_buttons = list()
 
 						if(!target_human.wear_mask)
 							target_buttons += "mouth"
+							if(target_human.client.prefs.read_preference(/datum/preference/toggle/erp/cum_face))
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]:
```suggestion
							if(target_human.client?.prefs.read_preference(/datum/preference/toggle/erp/cum_face))
```
Never assume a client to exist

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 2]
```diff
@@ -455,22 +455,24 @@
...
+								target_buttons += "face"
 
 						if(target_vagina && target_vagina?.is_exposed())
 							target_buttons += "vagina"
 
 						if(target_anus && target_anus?.is_exposed())
 							target_buttons += "asshole"
 
-						if(target_penis && target_penis?.is_exposed() && target_penis.sheath)
+						if(target_penis && target_penis?.is_exposed() && target_penis.sheath != "None")
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]:
```suggestion
						if(target_penis && target_penis.is_exposed() && target_penis.sheath != "None")
```
I know this isn't added in this PR but may as well avoid a double check here

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 3]
```diff
@@ -485,6 +487,17 @@
 							create_cum_decal = TRUE
 							visible_message(span_userlove("[src] shoots their sticky load onto the [target_human]!"), \
 								span_userlove("You shoot string after string of hot cum onto [target_human]!"))
+						else if(climax_into_choice == "face")
+							visible_message(span_userlove("[src] shoots their sticky load onto the [target_human] face!"), \
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]:
```suggestion
							visible_message(span_userlove("[src] shoots their sticky load onto [target_human]'s face!"), \
```

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 4]
```diff
@@ -485,6 +487,17 @@
...
+								span_userlove("You shoot string after string of hot cum onto [target_human] face!"))
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]:
```suggestion
								span_userlove("You shoot string after string of hot cum onto [target_human]'s face!"))
```

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 5]
```diff
@@ -485,6 +487,17 @@
...
+							if(target_human.GetComponent(/datum/component/cumfaced))
+								if(!target_human.GetComponent(/datum/component/cumfaced/big))
+									target_human.TakeComponent(/datum/component/cumfaced)
+									target_human.AddComponent(/datum/component/cumfaced/big)
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]: I'll have to look into this more to give a better suggestion but I'm instinctively not a fan of the double `GetComponent`

Comment (AnywayFarus) [reply modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]: i think i found a solution

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 6]
```diff
@@ -751,7 +765,7 @@
 //you got cum on your face bro *licks it off*
 /datum/component/cumfaced
 	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
-
+	var/big_load = 0
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]: true/false defines

File: modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm [hunk 7]
```diff
@@ -803,12 +817,13 @@
...
-		qdel(src)
-		return COMPONENT_CLEANED
+		return
+	qdel(src)
+	return COMPONENT_CLEANED
 
 /datum/component/cumfaced/big
 	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
-
+	big_load = 1
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm]: ditto

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Pull request merged**